rdtools/soiling.py

```diff
@@ -1052,3 +1065,1532 @@ def monthly_soiling_rates(soiling_interval_summary, min_interval_length=14,
+        consists of a degradation trend, a seasonal component, and a soiling
+        signal (defined as 1 if no soiling, decreasing with increasing soiling
+        losses). I.e.: PI = degradation_trend * seasonal_component * soiling_ratio *
+        residuals, or:
+
+        .. math::
+
+            PI = Rd * SC * SR * R
+
+        The function has a huristic for detecting whether the soiling signal is
+        significant enough for soiling loss inference, which is based on the
+        ratio between the spread in the soiling signal versus the spread in the
+        residuals (defined by the 2.5th and 97.5th percentiles)
+
+        * The degradation trend is obtained using the native RdTools Year-On-Year
+          method [1]_.
+        * The seasonal component is derived with statsmodels STL [2]_.
+        * The soiling signal is derived with a Kalman Filter with a cleaning
+          detection heuristic [3]_.
+
+        Parameters
+        ----------
+        order : tuple, defualt ('SR', 'SC', 'Rd')
+            Tuple containing 1 to 3 of the following strings 'SR' (soiling
+            ratio), 'SC' (seasonal component), 'Rd' (degradation component),
+            defining the order in which these components will be found during
+            iterative decomposition
+        degradation_method : string, default 'YoY'
+            Either 'YoY' or 'STL'. If anything else, 'YoY' will be assumed.
+            Decides whether to use the YoY method [3] for estimating the
+            degradation trend (assumes linear trend), or the STL-method (does
+            not assume linear trend). The latter is slower.
+        max_iterations : int, default 18
+            Max number of iterations to perform. Each iteration fits only one of the
+            components, so three iterations are needed to fit all three components.
+        detection_tuner : float, default .5
+            Higher value gives lower cleaning event detection sensitivity.
+            Should be between 0.1 and 2
+        convergence_criterium : float, default 5e-3
+            the relative change in the convergence metric required for
+            convergence
+        pruning_iterations : int, default 1
+        pruning_tuner : float, default .6
+            Should be between 0.1 and 2
+        soiling_significance_knob float, defualt 0.75
+        process_noise : float, default 1e-4
+        renormalize_SR : float, default None
+            If not none, defines the percentile for which the SR will be
+            normalized to, based on the SR just after cleaning events
+        ffill : bool, default True
+            Whether to use forward fill (default) or backward fill before
+            doing the rolling median for cleaning event detection
+        clip_soiling : bool, default True
+            Whether or not to clip the soiling ratio at max 1 and minimum 0.
+        verbose : bool, default False
+            If true, prints a progress report
+        ...
+
+        Returns
```

Comment:
Seven return values is quite a lot.  We should think about potential alternatives, for example a dictionary or some kind of `Result` object?

Reply:
That's true. Perhaps we just replace the whole list with a dictionary?